libcodechecker/analyze/analysis_manager.py

```diff
@@ -431,6 +464,26 @@ def check(check_data):
         return 1, skipped, reanalyzed, action.analyzer_type, None
 
 
+def create_actions_map(actions):
+    """
+    Create a dict for the build actions.
+    Key of the dict: (source_file, target).
```

Comment:
Is this a feasible solution in the future?
What if we compiler the same file twice but with different defines or once for shared object and once for static library but for the same target.

Reply:
We had a discussion with @dkrupp and @whisperity about this, and there is no better general solution we could provide atm.

In the future we might support different defines, we might have a key e.g. based on the output flag (-o). But that still would not enough to support special builds where an object is erased then recreated (Linux kernel).

Fortunately `-fpic` and its friends affect only the CodeGen but not the AST, so we can safely choose the entry either for the .so or for the .a .